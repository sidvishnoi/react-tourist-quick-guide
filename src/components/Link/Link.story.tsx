import { color, object, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import Link from '.';

const stories = storiesOf('Atoms/Link', module);
stories.addDecorator(withKnobs);

stories.add('with plain text', () => {
  return (
    <Link color={color('color', '#333')} href={text('href', '#')}>
      {text('text', 'hello')}
    </Link>
  );
});
