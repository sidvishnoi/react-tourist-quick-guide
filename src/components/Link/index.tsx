import * as React from 'react';
import styled from 'styled-components';

const Link = styled.a<{ color: string }>`
  text-decoration: none;
  color: ${props => props.color};

  :hover,
  :focus {
    text-decoration: underline;
  }

  @media print {
    color: #000;
  }
`;

export interface LinkProps {
  href: string;
  children: any;
  color?: string;
  [prop: string]: any;
}

export default function(props: LinkProps) {
  const { color = '#333', href, children, ...rest } = props;
  return (
    <Link color={color} href={href} {...rest}>
      {children}
    </Link>
  );
}
