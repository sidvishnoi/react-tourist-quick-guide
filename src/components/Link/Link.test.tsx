import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Link from '.';

describe('Components - Link', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Link href="LINK">TEXT</Link>);
    expect(tree).toMatchSnapshot();
  });

  it('renders <a> element', () => {
    const output = mount(<Link href="LINK">TEXT</Link>);

    expect(output.getDOMNode().nodeName).toEqual('A');
    expect(output.text()).toEqual('TEXT');
    expect(output.prop('href')).toEqual('LINK');
  });

  it('uses default color', () => {
    const output = mount(<Link href="LINK">TEXT</Link>);
    expect(output).toHaveStyleRule('color', '#333');
  });

  it('allows overriding props', () => {
    const output = mount(
      <Link href="LINK" color="red" role="button">
        TEXT
      </Link>,
    );
    expect(output).toHaveStyleRule('color', 'red');
    expect(output.prop('role')).toEqual('button');
  });
});
