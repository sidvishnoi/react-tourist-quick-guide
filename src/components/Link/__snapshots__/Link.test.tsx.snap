// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Link matches snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media print {
  .c0 {
    color: #000;
  }
}

<a
  className="c0"
  href="LINK"
>
  TEXT
</a>
`;
